base/db/R/clone_pft.R

```diff
@@ -0,0 +1,105 @@
+##' Duplicate existing pft with associated priors, species, and cultivars
+##'
+##' Creates a new pft that is a duplicate of an existing pft,
+##' including relationships with priors, species, and cultivars (if any) of the existing pft.
+##' This function mimics the 'clone pft' button in the PFTs record view page in the 
+##' BETYdb web interface for PFTs that aggregate >=1 species, but adds the ability to 
+##' clone the cultivar associations.
+##'
+##' @param parent.pft.name name of PFT to duplicate
+##' @param new.pft.name name for new PFT. Must not be the same as parent.pft.name
+##' @param new.pft.definition text for the new PFT's definition field.
+##' @param settings PEcAn settings list, used only for BETYdb connection parameters
+##' @return ID of the newly created pft in database, creates new PFT as a side effect
+##' @author David LeBauer, Chris Black
+##' @examples \dontrun{
+##' clone_pft(parent.pft.name    = "tempdecid",
+##'           new.pft.name       = "mytempdecid",
+##'           new.pft.definition = "mytempdecid is a new pft",
+##'           settings = pecan_settings_list)
+##' }
+clone_pft <- function(parent.pft.name,
+                      new.pft.name,
+                      new.pft.definition,
+                      settings){
+
+  if (new.pft.name == parent.pft.name) {
+    PEcAn.logger::logger.severe("new.pft.name must not be the same as parent.pft.name")
+  }
+
+  if (length(new.pft.name) > 1 || length(parent.pft.name) > 1) {
+    PEcAn.logger::logger.severe("multiple PFT names given, and clone_pft only knows how to handle one at a time.")
+  }
```

Comment:
A little while ago, I added functions `PEcAn.logger::logifnot/.../severeifnot`. Definitely no need to change this, but I'm curious what you think about that approach? I thought it might be a more concise way to do this frequent construct, but it might not be in practice:

```r
PEcAn.logger::severeifnot(
    length(new.pft.name) > 1 || length(parent.pft.name) > 1,
    msg = "Multiple PFT names given...
)
```

Reply:
I completely missed that you'd added these! I like them for clarity, even if it is verbose like all the logger calls are